super_csv/serializers.py

```diff
@@ -0,0 +1,30 @@
+"""
+Serializers for CSV operation data.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.contrib.auth.models import User
+from rest_framework import serializers
+
+from .models import CSVOperation
+
+
+class CSVOperationSerializer(serializers.ModelSerializer):
+    """
+    Serializer for history CSV bulk grade operations
+    """
+
+    user = serializers.SlugRelatedField(
+        slug_field='username',
+        queryset=User.objects.all()
+    )
+
+    class Meta:
+        model = CSVOperation
+        fields = ('class_name', 'unique_id', 'operation', 'user', 'modified')
+
+    @classmethod
+    def get_related_queryset(cls, queryset=None):
```

Comment:
Why is the serializer doing the query? It should just be serializing.

Reply:
I'm following a pattern Cale mentioned in response to an adjacent question I asked on slack: https://edx-internal.slack.com/archives/CDNLQM264/p1562075789005800?thread_ts=1562075435.005100&cid=CDNLQM264

> A pattern I introduced in course-discovery was to specifically have each serializer know how to prefetch its data, given the fields that it needs to serialize: https://github.com/edx/course-discovery/blob/master/course_discovery/apps/api/serializers.py#L229-L234

Reply:
I like this pattern because it colocates the reason you might want to alter the queryset in this way alongside the alteration, associating closely related ideas.

I understand it seems like it nominally violates single-responsibility, but the only reason this queryset is in there is for the efficiency of this serialization. They'd always be used together.

Reply:
Marking as resolved after removing the optional argument code path, since I agree that leaving open that option seems weird and doesn't really jive with the benefits of applying the pattern referenced above.